### DISCUSSION TITLE
Influxdb2 eats up 41G of memory, how to optimize it

### DISCUSSION THREAD

**Discussion by lifei64** [influxdata/influxdb#25020] [Q&A]
Let me talk about the environment first:
influxdb-version:2.0.7;
go-version:1.15.2;
Linux on physical machine: 64G of memory.

The problem is that now that 41G is used, I'm very worried that it will take up unlimited memory and eventually cause the process to crash. The following is relevant information. It occupies 66% of the memory. The actual memory allocated to it is about 13G. It has 3 buckets and a total of 70G files are occupied. And its buff/cache usage is very high. I understand that this is the index. Regarding the memory used by data, I want to limit the high memory usage of influxdb.

How to deal with it?
Please help me and Thanks very much!

**Comment by lifei64** [comment]
<img width="331" alt="image" src="https://github.com/influxdata/influxdb/assets/19682070/bece8830-75b6-4b3d-9ae7-11a8ee680ee1">
<img width="745" alt="image" src="https://github.com/influxdata/influxdb/assets/19682070/95425541-6822-450a-9d3b-52ad6ec662fe">

**Comment by raji-sridar** [comment]
Did you find any solutions, workarounds? We are facing the same issue as well.

**Reply by philjb** [reply]
A heap profile would be helpful to diagnose where the memory is being spent -- then we can make better recommendations:

https://go.dev/doc/diagnostics

I've had success with GOMEMLIMIT too. 

Let me know some more details.
